centreon/src/Core/ServiceTemplate/Application/UseCase/FindServiceTemplates/FindServiceTemplates.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ServiceTemplate\Application\UseCase\FindServiceTemplates;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\RequestParameters\Interfaces\RequestParametersInterface;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\ServiceTemplate\Application\Exception\ServiceTemplateException;
+use Core\ServiceTemplate\Application\Repository\ReadServiceTemplateRepositoryInterface;
+use Core\ServiceTemplate\Domain\Model\ServiceTemplate;
+
+final class FindServiceTemplates
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadServiceTemplateRepositoryInterface $repository,
+        private readonly RequestParametersInterface $requestParameters,
+        private readonly ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param FindServiceTemplatesPresenterInterface $presenter
+     */
+    public function __invoke(FindServiceTemplatesPresenterInterface $presenter): void
+    {
+        try {
+            if (
+                ! $this->user->isAdmin()
+                &&
+                (
```

Comment:
php-cs-fixer
```suggestion
                && (
```

Reply:
$this->user->hasTopologyRole() always return true when user is admin